src/_util/is.mjs

```diff
@@ -61,6 +61,8 @@ export function isNativeBigIntTypedArray(value) {
  */
 function isArrayBuffer(value) {
   try {
+    // ArrayBuffers are never arrays
+    if (Array.isArray(value)) return false;
```

Comment:
Please use functions from primordials.mjs to protect against prototype pollutions as much as possible.

```suggestion
    if (ArrayIsArray(value)) return false;
```